### ISSUE TITLE
Parse Error when mocking Illuminate\Support\Collection after upgrade to php 7

### ISSUE THREAD

**Issue by halaei** [padraic/mockery#530]
I have just upgraded from php 5.6 to php 7 and I find tests fails, all due to the doom try to mocking Laravel's Collection class. Here is a simple test that fails:

```php
class Php7Test extends \PHPUnit_Framework_TestCase
{
    public function test_foo()
    {
        \Mockery::mock(\Illuminate\Support\Collection::class);
    }
}
```
And here is the output:

1) Php7Test::test_foo
ParseError: syntax error, unexpected '.0' (T_DNUMBER), expecting ')'

    .../vendor/mockery/mockery/library/Mockery/Container.php:219
    .../vendor/mockery/mockery/library/Mockery.php:74
    .../tests/Unit/Php7Test.php:7

And here is a part of a generated code that has problem and can't be evaled by Mockery\Loader\EvalLoader::load()

```php
    public function flatten($depth = INF.0){
$argc = func_num_args();
$argv = func_get_args();
$ret = $this->_mockery_handleMethodCall(__FUNCTION__, $argv);
return $ret;
}
```
This code is, on my guess, trying to mock this function:

```php
    /**
     * Get a flattened array of the items in the collection.
     *
     * @param  int  $depth
     * @return static
     */
    public function flatten($depth = INF)
    {
        return new static(Arr::flatten($this->items, $depth));
    }
```

**Comment by GrahamCampbell** [comment]
> Illuminate\Support\Collection

Under what situation do you have to mock that class. I'd strongly recommend that you do not mock that class.

**Comment by GrahamCampbell** [comment]
Out of interest, does this work on `^1.0@dev` for the mockery version. I've had some issues with `^0.9` on php 7 in some of my test suites, and the 1.0 version of mockery has that stuff fixed.

**Comment by halaei** [comment]
I used this [github gist](https://gist.github.com/hollodotme/418e9b7c6ebc358e7fda) for upgrade. I think there is something wrong with php configs or it is machine dependent. Because when I installed php 7 on a fresh ubuntu 14.04 server on a virtual machine nothing this bad happened!

**Comment by halaei** [comment]
I agree that is not a good idea to mock that class. But again what is happening here?

**Comment by davedevelopment** [comment]
I have no idea what is happening here...

**Comment by halaei** [comment]
I cannot mock this as well:
```
class Php7Test extends \PHPUnit_Framework_TestCase
{
    public function test_foo()
    {
        \Mockery::mock(X::class);
    }
}
class X
{
    public function f($arg = INF)
    {
    }
}
```
Is that helpful?

**Comment by davedevelopment** [comment]
It doesn't help me any further, it runs fine for me:

``` php
<?php

use Mockery as m;

require "vendor/autoload.php";

class X
{
    public function f($arg = INF) {}
}

$mock = m::mock('X');                
var_dump(phpversion()); // string(5) "7.0.0"
var_dump($mock instanceof X); // bool(true)

```

**Comment by halaei** [comment]
Ok, I traced it to this point:
```php
var_export(INF);
```
And the output is: `INF.0`

I guess that is a big help!

**Comment by halaei** [comment]
Here is my PHP version iformation:

    PHP 7.0.2-1+deb.sury.org~trusty+1 (cli) ( NTS )
    Copyright (c) 1997-2015 The PHP Group
    Zend Engine v3.0.0, Copyright (c) 1998-2015 Zend Technologies
        with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies

I see var_export() [has changed in version 7.0.2](http://php.net/ChangeLog-7.php)! Are you using 7.0.2 as well?

**Comment by GrahamCampbell** [comment]
This looks very much like it's only broken in php 7.0.2+.

**Comment by halaei** [comment]
Can you please kindly report that? I am not comfortable with php.net bug report page!

**Comment by halaei** [comment]
Never mind. Done

**Comment by davedevelopment** [comment]
@halaei please could you link to the PHP bug report so we can track it?

**Comment by GrahamCampbell** [comment]
It's not a PHP bug?

**Comment by GrahamCampbell** [comment]
Or rather, the bug is fixed now...

**Comment by GrahamCampbell** [comment]
Fixing the bug broke mockery.

**Comment by davedevelopment** [comment]
I'm confused. My understanding is that var_export should return something parseable, but we're saying that in php 7.0.2 `var_export(INF)` is returning `INF.0`, which isn't parseable?

**Comment by halaei** [comment]
https://bugs.php.net/bug.php?id=71314
Have you reproduced the bug yet?

**Comment by xiphiaz** [comment]
@halaei also reproduced - php 7.0.2 on both travisci and official docker container. Probably best to just wait til 7.0.3 is out.

**Comment by adamwathan** [comment]
It was indeed a PHP bug, newly introduced when trying to fix a different bug. The fix for PHP 7.0.3 is here for reference: https://github.com/php/php-src/commit/ae50a0c0beba08e6abbc6cb5131fc44d5c6dde42

**Comment by GrahamCampbell** [comment]
So a bug introduced while fixing a bug, lol. ;)
